token/cli/src/main.rs

```diff
@@ -2105,6 +2110,27 @@ fn main() {
                 .about("Cleanup unnecessary token accounts")
                 .arg(owner_keypair_arg())
         )
+        .subcommand(
+            SubCommand::with_name("sync-native")
+                .about("Sync a native SOL token account to its underlying lamports")
+                .arg(
+                    owner_address_arg()
+                        .index(1)
+                        .conflicts_with("address")
```

Comment:
Should these really conflict?  It doesn't seem unreasonable for someone to have an aux account owned by a keypair other than the default wallet

Reply:
I was going off of the other commands like `account-info`, where you can specify a token + owner OR just address.  Here, since the token is always the native mint, the idea is you can specify owner OR address.  So if you specify "owner" it goes for the ATA, and if you specify "address" it goes straight for that address, which is why they would conflict.  If it's an aux account, do they typically just go for "address"?

I'm not sure how prevalent ATAs are for native token accounts, but the concept is to make it work like other mints.

Reply:
The difference is that `account-info` is just a cluster query, so `owner` doesn't need to sign a TX.  `transfer` is a (marginally) better example in this case

Reply:
Hm, there's something I'm not getting here.  Since `sync-native` is a permissionless instruction, the only required signer is the fee-payer, which to me is very similar to a cluster query.

Going back to your specific point maybe, how should the token account address be resolved if `address` and `owner` are specified?

Reply:
Ah!  I'd overlooked that it was permissionless.  Carry on :wink: